README.md

```diff
@@ -13,3 +13,33 @@
   complexity suggests a different time.
 
 Add your answers to this markdown file.
+
+Bullet 1
+
+1. Asymptotic analysis is theoretical, as it describes algorithm behavior with a set of unspecific inputs.  Actual performance is empirical, as it describes algorithm behavior with a specific set of inputs.  The two are different, as actual performance depends on constants and machine performance, whereas asymptotic analysis ignores those factors.
+
+
+2. Actual performance is affected by hardware, software, and the characteristics of the input data, whereas asymptotic analysis is primarily affected by the input size.
+
+3. In addition to ignoring constants, asymptotic analysis also ignores lower-order terms, which can be misleading when compared to actual performance since when lower-order terms are accounted for, they can have a decent effect on the runtime.
+
+Bullet 2
+
+The time complexity depends on if the binary search tree is balanced or not.  If it is balanced, it has a best case of O(1) and an average and worst case of O(log(n)).  If it isn’t balanced, it has a best case of O(1), and an average and worst case of O(n).
+
+For a balanced binary tree, finding the same element in a tree of 10,000 elements takes about 6.65 seconds in the worst case, which comes from multiplying 5 by log2(10,000)/log2(1,000).  For an unbalanced binary tree, finding the same element in a tree of 10,000 elements takes about 50 seconds in the worst case, which comes from multiplying 5 by 10,000/1,000. 
+
+Bullet 3
+
+1. It could have taken much longer because the tree was very unbalanced, meaning the algorithm has to go through most, if not all of the nodes.  This causes a large difference, as with a very unbalanced tree, the search time can approach O(n) instead of O(logn).  This alone doesn’t fully account for the large jump from 5 seconds to 100 seconds.  Additional overheads such as poor memory allocation and/or the use of an inefficient compiler combined with the very unbalanced tree could make the time take significantly longer.  Different compilers handle code differently, some more efficient than others, so if an inefficient compiler were used in this situation, the way the code is run and memory is stored could cause the runtime to take significantly longer when combined with poor memory allocation and an unbalanced tree.
```

Comment:
Not sure if this makes sense -- why would I want to use an inefficient compiler? And why would a compiler be inefficient for something as simple as binary search?